core/unit_test/TestTeamPolicyConstructors.hpp

```diff
@@ -121,11 +139,37 @@ void test_run_time_parameters() {
   ASSERT_EQ(p8.team_size(), team_size);
   ASSERT_EQ(p8.chunk_size(), chunk_size);
   ASSERT_EQ(p8.scratch_size(0), size_t(scratch_size));
+
+  Policy p9(league_size, Kokkos::AUTO);
+  ASSERT_EQ(p9.league_size(), league_size);
+#ifdef KOKKOS_ENABLE_THREADS
+  // team_size for Threads is -1 here
+  if (!std::is_same_v<ExecutionSpace, Kokkos::Threads>) {
+    ASSERT_GT(p9.team_size(), 0);
+  }
+#endif
```

Comment:
`team_size` returns -1 with Threads. Should this be considered a bug?

Reply:
Sounds like a bug. We should get a sensible `team_size` when using `Kokkos::AUTO`.